misk-redis/src/main/kotlin/misk/redis/RealRedis.kt

```diff
@@ -208,124 +187,137 @@ class RealRedis(private val jedisPool: JedisPool) : Redis {
-    }
+    val keyBytes = key.toByteArray(charset)
+    return jedis { expire(keyBytes, seconds) == 1L }
   }
 
   override fun expireAt(key: String, timestampSeconds: Long): Boolean {
-    return jedisPool.resource.use { jedis ->
-      jedis.expireAt(key, timestampSeconds) == 1L
-    }
+    val keyBytes = key.toByteArray(charset)
+    return jedis { expireAt(keyBytes, timestampSeconds) == 1L }
   }
 
   override fun pExpire(key: String, milliseconds: Long): Boolean {
-    return jedisPool.resource.use { jedis ->
-      jedis.pexpire(key, milliseconds) == 1L
-    }
+    val keyBytes = key.toByteArray(charset)
+    return jedis { pexpire(keyBytes, milliseconds) == 1L }
   }
 
   override fun pExpireAt(key: String, timestampMilliseconds: Long): Boolean {
-    return jedisPool.resource.use { jedis ->
-      jedis.pexpireAt(key, timestampMilliseconds) == 1L
-    }
+    val keyBytes = key.toByteArray(charset)
+    return jedis { pexpireAt(keyBytes, timestampMilliseconds) == 1L }
   }
 
   override fun watch(vararg keys: String) {
-    jedisPool.resource.use { jedis ->
-      jedis.watch(*keys)
-    }
+    val keysAsBytes = keys.map { it.toByteArray() }.toTypedArray()
+    jedisPool.resource.use { jedis -> jedis.watch(*keysAsBytes) }
   }
 
   override fun unwatch(vararg keys: String) {
-    jedisPool.resource.use { jedis ->
-      jedis.unwatch()
-    }
+    jedisPool.resource.use { jedis -> jedis.unwatch() }
   }
 
+  // Transactions do not get client histogram metrics right now.
+  // multi() returns the jedis to the pool, despite returning a Transaction that holds a reference.
+  // This is a bug, and will be fixed in a follow-up.
   override fun multi(): Transaction {
-    jedisPool.resource.use { jedis ->
-      return jedis.multi()
-    }
+    return jedisPool.resource.use { jedis -> jedis.multi() }
   }
 
+  // Pipelined requests do not get client histogram metrics right now.
+  // pipelined() returns the jedis to the pool, despite returning a Pipeline that holds a reference
+  // to the borrowed jedis connection.
+  // This is a bug, and will be fixed in a follow-up.
   override fun pipelined(): Pipeline {
-    jedisPool.resource.use { jedis ->
-      return jedis.pipelined()
-    }
+    return jedisPool.resource.use { jedis -> jedis.pipelined() }
   }
 
   /** Closes the connection to Redis. */
   override fun close() {
     return jedisPool.close()
   }
 
+  // Gets a Jedis instance from the pool, and times the requested method invocations.
+  private fun <T> jedis(op: JedisBinaryCommands.() -> T): T {
+    return jedisPool.resource.use { jedisResource ->
+      val invocationHandler = JedisTimedInvocationHandler(jedisResource, clientMetrics)
+      val timedProxy = JedisBinaryCommands::class.cast(
+        Proxy.newProxyInstance(
+          ClassLoader.getSystemClassLoader(),
+          arrayOf(JedisBinaryCommands::class.java),
+          invocationHandler
+        )
+      )
+      timedProxy.op()
+    }
+  }
```

Comment:
I think you are using `JedisBinaryCommands.() -> T` to make the syntax look real nice, e.g. 

```
jedis { incr(keyBytes) }
```

just a question, but where is the `op` param in L296 being used/passed?

Reply:
You actually got it right in the context of this comment.

`op` is used on L307. I could've called it like `op(timedProxy)`, but since `op` is written as an extension on the receiver, I called it in the same style.